cpp/src/arrow/compute/api_scalar.h

```diff
@@ -48,6 +48,25 @@ struct ARROW_EXPORT ElementWiseAggregateOptions : public FunctionOptions {
   bool skip_nulls;
 };
 
+/// Options for var_args_join.
+struct ARROW_EXPORT JoinOptions : public FunctionOptions {
+  /// How to handle null values. (A null separator always results in a null output.)
+  enum NullHandlingBehavior {
+    /// A null in any input results in a null in the output.
+    EMIT_NULL,
+    /// Nulls in inputs are skipped.
+    SKIP,
+    /// Nulls in inputs are replaced with the replacement string.
+    REPLACE,
```

Comment:
Nit, but I think we should avoid ALL_CAPS because of potential conflicts with third-party macros. What do you think?

Reply:
I was trying to stay consistent with the existing enums. (Also see: the whole ML discussion…) 

If we reach a consensus there I'm happy to rename all the enums.

Reply:
There's some other things I noticed in ARROW-13025 like a toplevel enum (not enum class).

Reply:
Fair enough.

Reply:
Hmm, I think FunctionOptions classes and enums are recent enough that we may want to do a cleanup pass on them.

Reply:
(as a separate JIRA, of course!)

Reply:
ARROW-13093